src/sparseml/keras/utils/pytorch2keras.py

```diff
@@ -0,0 +1,347 @@
+
+
+def convert_pytorch_to_keras(
+    arch_key: str,
+    type_: str,
+    class_type: str,
+    layer_mapping: Dict[str, str],
+    output_dir: str,
+    imagenet_dir: str,
+):
+    model_dir = os.path.join(output_dir, "{}-{}".format(arch_key, type_))
+    if not os.path.exists(model_dir):
+        os.makedirs(model_dir)
+    model_file_path = os.path.join(model_dir, "model.h5")
+
+    # Convert and save
+    kr_model = _convert_model(arch_key, type_, class_type, layer_mapping)
+    kr_model.save(model_file_path)
+
+    # Export to ONNX
+    exporter = ModelExporter(kr_model, model_dir)
+    exporter.export_onnx(name="model.onnx", debug_mode=False)
+
+    # Samples
+    samples_dir = os.path.join(model_dir, "samples")
+    if not os.path.exists(samples_dir):
+        os.makedirs(samples_dir)
+    val_dataset = ImageNetDataset(IMAGENET_DIR, train=False)
+    val_dataset = val_dataset.build(batch_size=100, repeat_count=1)
+    for img_batch, label_batch in val_dataset.take(1):
+        output_batch = kr_model(img_batch)
+        np.save(os.path.join(samples_dir, "inputs.npy"), img_batch)
+        np.save(os.path.join(samples_dir, "outputs.npy"), output_batch)
+        np.save(os.path.join(samples_dir, "labels.npy"), label_batch)
+
+
+def convert_resnets_for_keras(output_dir: str):
+    resnet_mapping = {
+        "input.conv": "input.conv",
+        "input.bn": "input.bn",
+        "sections.([0-9]+).([0-9]+).conv([0-9]+)": "sections.{}.{}.conv{}",
+        "sections.([0-9]+).([0-9]+).bn([0-9]+)": "sections.{}.{}.bn{}",
+        "sections.([0-9]+).([0-9]+).identity.conv": "sections.{}.{}.identity.conv",
+        "sections.([0-9]+).([0-9]+).identity.bn": "sections.{}.{}.identity.bn",
+        "classifier.fc": "classifier.fc",
+    }
+    convert_pytorch_to_keras(
+        "resnet50",
+        "base-none",
+        "single",
+        resnet_mapping,
+        output_dir,
+        IMAGENET_DIR,
+    )
+    convert_pytorch_to_keras(
+        "resnet50",
+        "pruned-conservative",
+        "single",
+        resnet_mapping,
+        output_dir,
+        IMAGENET_DIR,
+    )
+    convert_pytorch_to_keras(
+        "resnet50",
+        "pruned-moderate",
+        "single",
+        resnet_mapping,
+        output_dir,
+        IMAGENET_DIR,
+    )
+    convert_pytorch_to_keras(
+        "resnet101",
+        "base-none",
+        "single",
+        resnet_mapping,
+        output_dir,
+        IMAGENET_DIR,
+    )
+    convert_pytorch_to_keras(
+        "resnet101",
+        "pruned-moderate",
+        "single",
+        resnet_mapping,
+        output_dir,
+        IMAGENET_DIR,
+    )
+    convert_pytorch_to_keras(
+        "resnet152",
+        "base-none",
+        "single",
+        resnet_mapping,
+        output_dir,
+        IMAGENET_DIR,
+    )
+    convert_pytorch_to_keras(
+        "resnet152",
+        "pruned-moderate",
+        "single",
+        resnet_mapping,
+        output_dir,
+        IMAGENET_DIR,
+    )
+
+
+def main():
```

Comment:
can we add in some args and make the function callable for the key and weights to use? This would be instead of hardcoding the conversions into the script as that won't scale long term